src/main/kotlin/com/noumenadigital/npl/cli/service/SourcesManager.kt

```diff
@@ -0,0 +1,53 @@
+package com.noumenadigital.npl.cli.service
+
+import com.noumenadigital.npl.cli.exception.CommandExecutionException
+import com.noumenadigital.npl.lang.Source
+import java.io.File
+import java.nio.file.Files
+
+class SourcesManager(
+    private val projectDirectoryPath: String,
+) {
+    companion object {
+        private const val NPL_EXTENSION = ".npl"
+    }
+
+    private val testDirectory: String = "$projectDirectoryPath/src/test/npl"
```

Comment:
Let's see if we can avoid hardcoding this and not make any assumptions about the user's folder structure